### ISSUE TITLE
KeyError: 'assembly_accession'

### ISSUE THREAD

**Issue by scorreard** [kblin/ncbi-genome-download#209]
Hi!

Version :
```
ncbi-genome-download --version
0.3.1
```

Command :
```
ncbi-genome-download \
    -s genbank -A GCA_029448725.1 --formats fasta all \
     \
    --output-folder ./ \
    --flat-output
```

Error:
```
Traceback (most recent call last):
  File "/usr/local/bin/ncbi-genome-download", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/__main__.py", line 26, in main
    ret = args_download(args)
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/core.py", line 178, in args_download
    return config_download(config)
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/core.py", line 197, in config_download
    download_candidates = select_candidates(config)
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/core.py", line 308, in select_candidates
    for entry in filter_entries(entries, config):
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/core.py", line 354, in filter_entries
    if not config.is_compatible_assembly_accession(entry['assembly_accession']):
KeyError: 'assembly_accession'
```

Very similar to this older issue : https://github.com/kblin/ncbi-genome-download/issues/2

**Comment by chasemc** [comment]
Same

**Comment by zxyEmily** [comment]
Same

**Comment by andrewjpage** [comment]
And same here

**Comment by cpbivins28** [comment]
Same

**Comment by Ikajiro** [comment]
I got the same issue on the script installed more than one year before. There was no issue when I tried it about half an year   before. The change I had after that is the upgrade of macOS to Ventura recently.

**Comment by dnolin13** [comment]
also getting this issue

**Comment by andrewjpage** [comment]
Pull request https://github.com/kblin/ncbi-genome-download/pull/210 fixes this issue. While you are waiting for an official release you can just edit the few lines of code to correct the error.

**Comment by SharonYXiao** [comment]
Pull request #210 didn't work for us. But thanks to andrewjpage for the tip, this is a issue with the format of the 'assembly_summary_genbank.txt' file. Specifically, the header columns for '# assembly_accession' has changed to '#assembly_accession'. As a result, the code in 'core.py' that references the 'entry' dictionary is no longer accurate.

To resolve this issue, I have replaced the old code 'entry['assembly_accession']' with the updated code 'entry['#assembly_accession']'. This change was made on whole script including line 354, which was the location of the reported error. After making this change, I was able to successfully download the genome without any further issues.

**Comment by chasemc** [comment]
Those having issues installing a clean version of the PR can try:
https://github.com/kblin/ncbi-genome-download/pull/210#issuecomment-1591699825

**Comment by nick-youngblut** [comment]
Once v0.3.1 is installed in a conda env, it appears that the error then persists, regardless of downgrading to a version previous to v0.3.1. One has to create a new conda env just to get around the issue.

**Comment by Joshua-Macleod** [comment]
> 

I created a new env and installed v0.3.0 and the same issue arose, albeit on different lines:

`Traceback (most recent call last):
  File "/home/ubuntu/mambaforge/envs/miscold/bin/ncbi-genome-download", line 10, in <module>
    sys.exit(main())
  File "/home/ubuntu/mambaforge/envs/miscold/lib/python3.9/site-packages/ncbi_genome_download/__main__.py", line 26, in main
    ret = args_download(args)
  File "/home/ubuntu/mambaforge/envs/miscold/lib/python3.9/site-packages/ncbi_genome_download/core.py", line 174, in args_download
    return config_download(config)
  File "/home/ubuntu/mambaforge/envs/miscold/lib/python3.9/site-packages/ncbi_genome_download/core.py", line 193, in config_download
    download_candidates = select_candidates(config)
  File "/home/ubuntu/mambaforge/envs/miscold/lib/python3.9/site-packages/ncbi_genome_download/core.py", line 290, in select_candidates
    for entry in filter_entries(entries, config):
  File "/home/ubuntu/mambaforge/envs/miscold/lib/python3.9/site-packages/ncbi_genome_download/core.py", line 336, in filter_entries
    if not config.is_compatible_assembly_accession(entry['assembly_accession']):
KeyError: 'assembly_accession'
`

**Comment by Joshua-Macleod** [comment]
> One has to create a new conda env just to get around the issue.

Created a new conda env and installed v0.3.0 and the issue persists, albeit different lines: 

```
Traceback (most recent call last):
  File "/usr/local/bin/ncbi-genome-download", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/__main__.py", line 26, in main
    ret = args_download(args)
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/core.py", line 178, in args_download
    return config_download(config)
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/core.py", line 197, in config_download
    download_candidates = select_candidates(config)
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/core.py", line 308, in select_candidates
    for entry in filter_entries(entries, config):
  File "/usr/local/lib/python3.10/site-packages/ncbi_genome_download/core.py", line 354, in filter_entries
    if not config.is_compatible_assembly_accession(entry['assembly_accession']):
KeyError: 'assembly_accession'
```

**Comment by Joshua-Macleod** [comment]
Oh curious, definitely installed via mamba into an env - have subsequently had success following below as described above:

> To resolve this issue, I replaced the old code 'entry['assembly_accession']' with the updated code 'entry['#assembly_accession']'. This change was made on whole script including line 354, which was the location of the reported error. After making this change, I was able to download the genome without any further issues.

**Comment by caligarimga** [comment]
> Pull request #210 didn't work for me because the installation failed. But thanks to andrewjpage for the tip, this is an issue with the format of the 'assembly_summary_genbank.txt' file. Specifically, the header columns for '# assembly_accession' was changed to '#assembly_accession'. As a result, the code in 'core.py' that references the 'entry' dictionary is no longer accurate.
> 
> To resolve this issue, I replaced the old code 'entry['assembly_accession']' with the updated code 'entry['#assembly_accession']'. This change was made on whole script including line 354, which was the location of the reported error. After making this change, I was able to download the genome without any further issues.

It's worked here, thank you!

**Comment by alexmaehon** [comment]
in google colab get 'assembly_accesion' error: 
"KeyError                                  Traceback (most recent call last)
[<ipython-input-6-179a325d4a32>](https://localhost:8080/#) in <cell line: 2>()
      1 import ncbi_genome_download as ngd
----> 2 ngd.download()

3 frames
[/usr/local/lib/python3.10/dist-packages/ncbi_genome_download/core.py](https://localhost:8080/#) in filter_entries(entries, config)
    352                          entry['taxid'], config.taxids)
    353             continue
--> 354         if not config.is_compatible_assembly_accession(entry['assembly_accession']):
    355             logger.debug('Skipping entry with incompatible assembly accession %r', entry['assembly_accession'])
    356             continue

KeyError: 'assembly_accession'
"

**Comment by kblin** [comment]
Fixed by bd4607102748673c2f2e45719213b74fda37b606 included in the new 0.3.2 release now on pypi. Bioconda packages watch the pypi changes and should be available soon.
